src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EmptyStatementCheck.java

```diff
@@ -40,6 +40,24 @@
  * <pre>
  * &lt;module name=&quot;EmptyStatement&quot;/&gt;
  * </pre>
+ * <p>Example:</p>
+ * <div class="wrapper">
+ * <pre>
+ * class Expression {
+ *     List&lt;Expression&gt; arguments;
+ *     public void calculate() {
+ *         // Calculates something
+ *     }
+ *
+ *     public bool isConstant() {
+ *         // Returns true, if Expression is constant, else returns false
+ *     }
+ * }
+ * for (int i = 0; i &lt; arguments.size(); arguments.get(i++).calculate()); // OK
+ * if (arguments.get(0).isConstant()); // Can violate NullPointerException if arguments or arguments[0] is null,
```

Comment:
No need for such long description, `// violation` is enough. Please add more examples